### PR TITLE
Team subclass

### PR DESCRIPTION
**Pull request by EHale64**
Final commit and merge fixes

### CODE REVIEW

File: lib/calculable.rb [hunk 1]
```diff
@@ -4,5 +4,9 @@ def average(numerator, denominator)
     (numerator.to_f/denominator).round(2)
   end
 
-
+  def calculate_win_percentage(team_games)
+    wins = 0.0
+    team_games.each {|game| wins += 1.0 if game.result == "WIN"}
+    (wins / team_games.length)
+  end
```
Comment (jpc20) [inline lib/calculable.rb:11]: Did we test this?

File: lib/team_stats.rb [hunk 2]
```diff
@@ -1,5 +1,64 @@
 require_relative "stats"
-
 class TeamStats < Stats
 
+ def team_info(team_id)
+   team = team_by_id(team_id)
+    team_data = team.instance_variables.map { |key,value| ["#{key}".delete("@"), value = team.send("#{key}".delete("@").to_sym)]}.to_h
```
Comment (EHale64) [inline lib/team_stats.rb:6]: Returns an array of the team object's instance variables, then iterates over that array, deletes the '@' from the front of the instance variable and assigns that as a key, then sets the value equal to the key by again removing the '@' and then passing that as a method call then returning the key, value pairs as a hash.

File: lib/team_stats.rb [hunk 3]
```diff
@@ -1,5 +1,64 @@
...
+    team_data.delete_if {|k,v| k == "stadium"}
```
Comment (EHale64) [inline lib/team_stats.rb:7]: Searches the hash for a key, value pair whose key is "stadium" then deletes it.

File: lib/team_stats.rb [hunk 4]
```diff
@@ -1,5 +1,64 @@
...
+ end
+
+ def best_season(team_id)
+   team_seasons = @games.find_all {|game| game.home_team_id  == team_id || game.away_team_id == team_id }
```
Comment (EHale64) [inline lib/team_stats.rb:11]: possible refactor for best and worst seasons

**Review by EHale64 — COMMENTED** [review]
Looks good minor instances for potential refactor

**Pull request merged**